app/src/main/java/org/mozilla/fenix/settings/quicksettings/QuickSettingsController.kt

```diff
@@ -136,7 +139,8 @@ class DefaultQuickSettingsController(
         val permissions = sitePermissions
 
         sitePermissions = if (permissions == null) {
-            val origin = requireNotNull(session?.url?.toUri()?.host) {
+            val tab = browserStore.state.findTabOrCustomTab(sessionId)
```

Comment:
Why not falling back to the selected session and avoid the `requireNotNull` after?

Reply:
Because even with `findTabOrCustomTabOrSelectedTab` you'd still need the null check as the return type is still nullable. And also because we're now only creating the controller with a `sessionId` we know exists.

Reply:
Got it, thanks for the clarification!